### DISCUSSION TITLE
Installing and running DREAM3DNX v 7.0.0-RC-12

### DISCUSSION THREAD

**Discussion by AmitS07** [BlueQuartzSoftware/DREAM3DNX-Issues#19] [Q&A]
Hi all, 
I downloaded the DREAM3DNX v 7.0.0-RC-12 for Linux system. But, not able to run the DREAM3DNX. Can anyone suggest what I need to do. I am new to this. I tried with the following command- (from bin directory)
` ./nxrunner -p ~/software/DREAM3DNX-7.0.0-RC-12-linux-x86_64/Pipeline.json`
I guess, this is not correct, because I am not able to find and "Pipeline" drive. 

Also, let me know the same for DREAM3D-6.5.171-Linux-x86_64.

Even, I am not able to find the documentation on how to run this from terminal.

Thanks

**Comment by imikejackson** [comment]
@AmitS07 My guess is that the $SHELL that you are using is not properly expanding the `~` character when passing that into the 'nxrunner' application. Can you send the complete output of the command?

**Reply by AmitS07** [reply]
> @AmitS07 My guess is that the $SHELL that you are using is not properly expanding the `~` character when passing that into the 'nxrunner' application. Can you send the complete output of the command?

Thanks @imikejackson for your quick response, here is full command-
`(base) asingh38@taylor:~/software/DREAM3DNX-7.0.0-RC-12-linux-x86_64/bin$ ./nxrunner -p /home/asingh38/software/DREAM3DNX-7.0.0-RC-12-linux-x86_64/Pipeline.json`

I tried with by giving full path, but did not work. 
The error , I got is this -
```
Could not load pipeline at path: '/home/asingh38/software/DREAM3DNX-7.0.0-RC-12-linux-x86_64/Pipeline.json'
Error -1: Failed to open file '/home/asingh38/software/DREAM3DNX-7.0.0-RC-12-linux-x86_64/Pipeline.json'
```

**Comment by nyoungbq** [comment]
Hello @AmitS07,

I looked at the input you have supplied and at first glance it looks correct, but the extension should be `.d3dpipeline` if it is a pipeline from NX. It aligns with the documentation which can be found [here](https://github.com/BlueQuartzSoftware/simplnx/blob/develop/src/nxrunner/docs/NX_CLI.md). I work primarily on a Linux machine, so I will attempt to recreate this locally. Did you attempt suppling an absolute path to the pipeline you are trying to execute?

Also to clarify, NXRunner is our CLI pipeline runner, the full DREAM3DNX project has an integrated UI used to create the pipelines in question. If your intention is to develop pipelines and visualize the data, I would recommend running the DREAM3DNX executable.

Edit:
I just downloaded and ran the executable locally to ensure it functions it should be: `./DREAM3DNX` from that same binary directory.

**Reply by AmitS07** [reply]
Hi @nyoungbq 
Yes, it is from NX, though I am not sure about the "Pipeline". I am not able to see any directory name pipeline inside "bin" directory. 
yes, after your comments I tried. But, did not work
No, I am not planning to develop pipelines. 
Please see the above comments to check the full-path. 

Just to notify, I am able to open the DREAM3DNX from Windows. 
Sorry that, I am totaly new to this, I have no idea what is going. Let me also try the same in older-version of DREAM3D
 Thanks

**Comment by imikejackson** [comment]
@AmitS07 If the pipeline file came from DREAM.3D version 6.x then those have a file extension of `.json` and from the looks of it (trying here locally), `nxrunner` (part of DREAM3D-NX) does not support reading in those older pipelines. You will first have to launch DREAM3DNX itself, import the .json file into DREAM3D-NX which should then try to do its best to convert the pipeline. You can then save the imported pipeline to a new `.d3dpipeline` file. Now that you have the .d3dpipeline file 'nxrunner' should be able to run that. 

Running on a MacOS system I get the following output:
```
1007:[mjackson@Octane:Applications]% ./DREAM3DNX.app/Contents/bin/nxrunner -p /Users/mjackson/Desktop/D3D_Pipelines/1_Jay_Lee.json 
nxrunner: Version 1.3.0 Build Date:2024/06/05

Loading Plugins from /Users/mjackson/Applications/DREAM3DNX.app/Contents/bin
Loading Plugins from /Users/mjackson/Applications/DREAM3DNX.app/Contents/Plugins
Loading Plugin: /Users/mjackson/Applications/DREAM3DNX.app/Contents/Plugins/libOrientationAnalysis.simplnx
Loading Plugin: /Users/mjackson/Applications/DREAM3DNX.app/Contents/Plugins/libSimplnxCore.simplnx
Loading Plugin: /Users/mjackson/Applications/DREAM3DNX.app/Contents/Plugins/libITKImageProcessing.simplnx
Could not load pipeline at path: '/Users/mjackson/Desktop/D3D_Pipelines/1_Jay_Lee.json'
Error -1: Pipeline JSON did not contain key 'name'
```

**Comment by imikejackson** [comment]
@AmitS07 nxrunner does have a conversion function already available to you by using `nxrunner --convert-output .....`

**Reply by AmitS07** [reply]
> @AmitS07 nxrunner does have a conversion function already available to you by using `nxrunner --convert-output .....`


@imikejackson This make sense. I guess, the command that I was using was for DREAM3D v-6. Though, I got some idea. I will try on this. But, I find DREAM3D setup is quite easy for windows rather than Linux.
 Thanks for the discussion.
